src/Engine/Data/TextureManager.cpp

```diff
@@ -141,17 +141,18 @@ void TextureManager::deleteTexture( Texture* texture ) {
 }
 
 void TextureManager::updateTextureContent( const std::string& texture, void* content ) {
-    CORE_ASSERT( m_textures.find( texture ) != m_textures.end(),
-                 "Trying to update non existing texture" );
     m_pendingData[texture] = content;
 }
 
 void TextureManager::updatePendingTextures() {
     if ( m_pendingData.empty() ) { return; }
 
     for ( auto& data : m_pendingData ) {
-        LOG( logINFO ) << "TextureManager::updateTextures \"" << data.first << "\".";
-        m_textures[data.first]->updateData( data.second );
+        auto itr = m_textures.find( data.first );
+        if ( itr != m_textures.end() ) { itr->second->updateData( data.second ); }
```

Comment:
I don't know if it's a good idea to make the function updatePendingTextures more complex (adding a condition) here because this function with the contribution of this PR will be called for each rendered frame. Would it be possible that the Texture::updateData function can initialize a texture with his textureParameters if it is not already initialized before affecting the image data without overcost ?

Reply:
Well, in any case you have to test for something (the test will be done only if pending texture is not empty, do every frame, only for updated texture ...)
Another way to do it is to unsure that updateTextureData is called after the texture is initialized.